Incorrect method argument types in generation.rs and validation.rs

### System Info

I cloned the repo and tried docker build (host machine Ubuntu 22.04)

The commit is:

`d2ae358, 3 weeks ago, Wed Aug 2 17:23:54 2023 -0700, Yang, Bo, bo@preemo.io, Claim copyright (#7)`

I proceeded to build:

`docker build -t text-generation-inference:d2ae358 .`

And I got this error 
```
 > [builder 10/10] RUN cargo build --release:
0.350 info: syncing channel updates for '1.70.0-x86_64-unknown-linux-gnu'
1.325 info: latest update on 2023-06-01, rust version 1.70.0 (90c541806 2023-05-31)
1.354 info: downloading component 'clippy'
1.647 info: downloading component 'rustfmt'
2.109 info: installing component 'clippy'
2.403 info: installing component 'rustfmt'
3.158    Compiling text-generation-client v0.9.4 (/usr/src/router/client)
3.161    Compiling grpc-metadata v0.1.0 (/usr/src/router/grpc-metadata)
3.164    Compiling text-generation-router v0.9.4 (/usr/src/router)
3.166    Compiling text-generation-launcher v0.9.4 (/usr/src/launcher)
5.821    Compiling text-generation-benchmark v0.9.4 (/usr/src/benchmark)
6.388 error[E0308]: mismatched types
6.388    --> benchmark/src/generation.rs:221:17
6.388     |
6.388 221 |         .decode(Vec::from(encoding.get_ids()), false)
6.388     |          ------ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
6.388     |          |      |
6.388     |          |      expected `&[u32]`, found `Vec<u32>`
6.388     |          |      help: consider borrowing here: `&Vec::from(encoding.get_ids())`
6.388     |          arguments to this method are incorrect
6.388     |
6.388     = note: expected reference `&[u32]`
6.388                   found struct `Vec<u32>`
6.388 note: method defined here
6.388    --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokenizers-0.13.4/src/tokenizer/mod.rs:814:12
6.388     |
6.388 814 |     pub fn decode(&self, ids: &[u32], skip_special_tokens: bool) -> Result<String> {
6.388     |            ^^^^^^
6.388 
6.429 For more information about this error, try `rustc --explain E0308`.
6.441 error: could not compile `text-generation-benchmark` (lib) due to previous error
6.441 warning: build failed, waiting for other jobs to finish...
6.989 error[E0308]: mismatched types
6.989    --> router/src/validation.rs:314:25
6.989     |
6.989 314 |                 .decode(Vec::from(encoding.get_ids()), false)
6.989     |                  ------ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
6.989     |                  |      |
6.989     |                  |      expected `&[u32]`, found `Vec<u32>`
6.989     |                  |      help: consider borrowing here: `&Vec::from(encoding.get_ids())`
6.989     |                  arguments to this method are incorrect
6.989     |
6.989     = note: expected reference `&[u32]`
6.989                   found struct `Vec<u32>`
6.989 note: method defined here
6.989    --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokenizers-0.13.4/src/tokenizer/mod.rs:814:12
6.989     |
6.989 814 |     pub fn decode(&self, ids: &[u32], skip_special_tokens: bool) -> Result<String> {
6.989     |            ^^^^^^
6.989 
7.055 error: could not compile `text-generation-router` (lib) due to previous error
------
Dockerfile:36
--------------------
  34 |     COPY router router
  35 |     COPY launcher launcher
  36 | >>> RUN cargo build --release
  37 |     
  38 |     # Python builder
--------------------
ERROR: failed to solve: process "/bin/sh -c cargo build --release" did not complete successfully: exit code: 101
￼
```




### Information

- [X] Docker
- [ ] The CLI directly

### Tasks

- [X] An officially supported command
- [ ] My own modifications

### Reproduction

Clone the repo, docker build.

### Expected behavior

Expected it to build without rust error

--- a/benchmark/src/generation.rs
+++ b/benchmark/src/generation.rs
@@ -218,6 +218,6 @@ fn create_sequence(sequence_length: u32, tokenizer: Tokenizer) -> String {
     encoding.truncate(sequence_length as usize, 0, TruncationDirection::Left);
     // Decode
     tokenizer
-        .decode(Vec::from(encoding.get_ids()), false)
+        .decode(encoding.get_ids().as_ref(), false)
         .unwrap()
 }

--- a/benchmark/src/generation.rs
+++ b/benchmark/src/generation.rs
@@ -218,6 +218,6 @@ fn create_sequence(sequence_length: u32, tokenizer: Tokenizer) -> String {
     encoding.truncate(sequence_length as usize, 0, TruncationDirection::Left);
     // Decode
     tokenizer
-        .decode(encoding.get_ids().as_ref(), false)
+        .decode(From::from(encoding.get_ids()), false)
         .unwrap()
 }

--- a/router/src/validation.rs
+++ b/router/src/validation.rs
@@ -311,7 +311,7 @@ fn prepare_input(
             // truncate encoding and decode new inputs
             encoding.truncate(truncate, 0, TruncationDirection::Left);
             let inputs = tokenizer
-                .decode(encoding.get_ids().as_ref(), false)
+                .decode(From::from(encoding.get_ids()), false)
                 .map_err(|err| ValidationError::Tokenizer(err.to_string()))?;
             (inputs, encoding.len())
         }

--- a/router/src/validation.rs
+++ b/router/src/validation.rs
@@ -311,7 +311,7 @@ fn prepare_input(
             // truncate encoding and decode new inputs
             encoding.truncate(truncate, 0, TruncationDirection::Left);
             let inputs = tokenizer
-                .decode(Vec::from(encoding.get_ids()), false)
+                .decode(encoding.get_ids().as_ref(), false)
                 .map_err(|err| ValidationError::Tokenizer(err.to_string()))?;
             (inputs, encoding.len())
         }

